lib/util.js

```diff
@@ -73,59 +73,109 @@ const {
 
 let internalDeepEqual;
 
+/**
+ * @param {any} arg
+ * @returns {arg is boolean}
+ */
 function isBoolean(arg) {
   return typeof arg === 'boolean';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is null}
+ */
 function isNull(arg) {
   return arg === null;
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is (null | undefined)}
+ */
 function isNullOrUndefined(arg) {
   return arg === null || arg === undefined;
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is number}
+ */
 function isNumber(arg) {
   return typeof arg === 'number';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is string}
+ */
 function isString(arg) {
   return typeof arg === 'string';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is symbol}
+ */
 function isSymbol(arg) {
   return typeof arg === 'symbol';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is undefined}
+ */
 function isUndefined(arg) {
   return arg === undefined;
 }
 
+/**
+ * @param {any} arg
+ * @returns {boolean}
+ */
 function isObject(arg) {
   return arg !== null && typeof arg === 'object';
 }
 
+/**
+ * @param {any} e
+ * @returns {arg is Error}
+ */
 function isError(e) {
   return ObjectPrototypeToString(e) === '[object Error]' || e instanceof Error;
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is Function}
+ */
 function isFunction(arg) {
   return typeof arg === 'function';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is (null | boolean | string | number | symbol)}
```

Comment:
Added `undefined` and sorted
```suggestion
 * @returns {arg is (boolean | null | number | string | symbol | undefined)}
```